packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -8193,6 +8193,29 @@ const testsTypescript = {
         },
       ],
     },
+    {
+      code: normalizeIndent`
+        function useCustomCallback(callback, deps) {
+          return useCallback(callback as any, deps)
+        }
+      `,
+      errors: [
+        {
+          message:
+            "React Hook useCallback has a missing dependency: 'callback'. Either include it or remove the dependency array.",
```

Comment:
This should rather warn that we can't analyze the dependencies of `callback`. Using `[callback]` in the dependencies would be equivalent to not using `useCallback` at all.

Reply:
Good idea, I updated the PR to use the unknown dependencies message when argument is an identifier passed as a function parameter.